learn/advanced/pagination.md

```diff
@@ -129,95 +140,124 @@ document.querySelector('#previous_button').onclick = function () { updatePageNum
 document.querySelector('#next_button').onclick = function () { updatePageNumber(this) }
 ```
 
-## Not recommended: Page selection
+## Numbered page selectors
 
-This type of pagination consists of a numbered list of pages accompanied by next and previous buttons.
+This type of pagination consists of a numbered list of pages accompanied by "Next" and "Previous" buttons. This is a common UI pattern that offers users a significant amount of precision when navigating results.
 
-This is a common UI pattern that offers users a significant amount of precision when navigating results. However, due to Meilisearch's [limitations](#choosing-the-right-pagination-ui), it is not a good fit for pagination with Meilisearch.
+Calculating the total amount of search results for a query is a resource-intensive process. Because of that, page selectors might lead to slower interfaces.
 
 ### Implementation
 
-The general implementation of a page selection interface is similar to our [recommended solution](#recommended-previous-and-next-buttons), with one significant addition: it includes a numbered page list.
+By default, Meilisearch queries only return `estimatedTotalHits`. This value is likely to change as a user navigates search results and should not be used to create calculate the number of search result pages.
+
+When your query contains either `hitsPerPage`, `page`, or both these search parameters, Meilisearch returns `totalHits` and `totalPages` instead of `estimatedTotalHits`. `totalHits` contains the exhaustive number of results for that query, and `totalPages` contains the exhaustive number of pages of search results for the same query:
```

Comment:
```suggestion
When your query contains either [`hitsPerPage`](/reference/api/search.md#number-of-results-per-page), [`page`](/reference/api/search.md#page), or both these search parameters, Meilisearch returns `totalHits` and `totalPages` instead of `estimatedTotalHits`. `totalHits` contains the exhaustive number of results for that query, and `totalPages` contains the exhaustive number of pages of search results for the same query:
```